src/main/scala/clump/Clump.scala

```diff
@@ -67,15 +67,6 @@ object Clump {
 
   def collect[T](clumps: List[Clump[T]]): Clump[List[T]] =
     new ClumpCollect(clumps)
-
-  def sourceFrom[T, U](fetch: Set[T] => Future[Map[T, U]], maxBatchSize: Int = Int.MaxValue) =
```

Comment:
I'm a bit on the fence about this. I really like what you've done and made things really tidy and readable. However I still think there is something to be said for being able to initiate everything via the `Clump` object - so the user does not need to know about two classes.

WDYT?

It would be 3 convenience method similar to what we had before that just call through:
```scala
def sourceFrom[T, U, C](fetch: C => Future[Map[T, U]])(implicit cbf: CanBuildFrom[Nothing, T, C]) = ClumpSource.from(fetch)
def source[T, U, C](fetch: C => Future[Iterable[U]])(keyExtractor: U => T)(implicit cbf: CanBuildFrom[Nothing, T, C]) = ClumpSource(fetch, keyExtractor)
def sourceZip[T, U](fetch: List[T] => Future[List[U]]): ClumpSource[T, U] = ClumpSource.zip(fetch)
```